x-pack/plugins/infra/public/components/asset_details/tabs/overview/full_page_view.tsx

```diff
@@ -0,0 +1,87 @@
+import { MetricsGrid } from './metrics/metrics_grid';
+import { useAssetDetailsRenderPropsContext } from '../../hooks/use_asset_details_render_props';
+import { useMetadataStateProviderContext } from '../../hooks/use_metadata_state';
+import { useDataViewsProviderContext } from '../../hooks/use_data_views';
+import { useDateRangeProviderContext } from '../../hooks/use_date_range';
+import { SectionSeparator } from './section_separator';
+
+export const FullPageView = () => {
+  const { dateRange } = useDateRangeProviderContext();
+  const { asset, assetType } = useAssetDetailsRenderPropsContext();
+  const {
+    metadata,
+    loading: metadataLoading,
+    error: fetchMetadataError,
+  } = useMetadataStateProviderContext();
+  const { logs, metrics } = useDataViewsProviderContext();
+
+  return (
+    <EuiFlexGroup direction="column" gutterSize="m">
+      <EuiFlexItem grow={false}>
+        <KPIGrid nodeName={asset.name} timeRange={dateRange} dataView={metrics.dataView} />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        {fetchMetadataError ? (
+          <EuiCallOut
+            title={i18n.translate('xpack.infra.assetDetailsEmbeddable.overview.errorTitle', {
+              defaultMessage: 'Sorry, there was an error',
+            })}
+            color="danger"
+            iconType="error"
+            data-test-subj="infraMetadataErrorCallout"
+          >
+            <FormattedMessage
+              id="xpack.infra.assetDetailsEmbeddable.overview.errorMessage"
+              defaultMessage="There was an error loading your host metadata. Try to {reload} and open the host details again."
+              values={{
+                reload: (
+                  <EuiLink
+                    data-test-subj="infraAssetDetailsMetadataReloadPageLink"
+                    onClick={() => window.location.reload()}
+                  >
+                    {i18n.translate('xpack.infra.assetDetailsEmbeddable.overview.errorAction', {
+                      defaultMessage: 'reload the page',
+                    })}
+                  </EuiLink>
+                ),
+              }}
+            />
+          </EuiCallOut>
+        ) : (
+          <MetadataSummaryList metadata={metadata} metadataLoading={metadataLoading} />
+        )}
+        <SectionSeparator />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <AlertsSummaryContent assetName={asset.name} assetType={assetType} dateRange={dateRange} />
+        <SectionSeparator />
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <MetricsGrid
+          timeRange={dateRange}
+          logsDataView={logs.dataView}
+          metricsDataView={metrics.dataView}
+          nodeName={asset.name}
+          showNginxStubstatus={
+            !!metadata?.features.find((feature) => feature.name === 'nginx.stubstatus')
+          }
+          showNginxAccess={!!metadata?.features.find((feature) => feature.name === 'nginx.access')}
```

Comment:
Thanks, I changed it to `some`. I like the idea of having a Map there. It will make this easier but currently, we have more than one key-value pair (it includes the feature's `source` not only the name) and that will mean modifying the response structure. Of course, we can check different options in the future on how to optimize this.